src/main/cpp/src/query_operations/broad_combined_gvcf.cc

```diff
@@ -308,7 +310,9 @@ BroadCombinedGVCFOperator::BroadCombinedGVCFOperator(VCFAdapter& vcf_adapter, co
                                          +callset_name+" to the combined VCF/gVCF header");
     }
   }
-  bcf_hdr_sync(m_vcf_hdr);
+  if (bcf_hdr_sync(m_vcf_hdr)) {
+    logger.error("bcf_hdr_sync() failed while {}:{}", "adding samples to template header", __LINE__);
+  }
```

Comment:
A quick look at bcf_hdr_sync shows that this error can occur only if there's a memory allocation failure - so maybe a short phrase about realloc() failed and throw an exception.